lib/stitches/auto_fill.py

```diff
@@ -198,11 +210,22 @@ def insert_node(graph, shape, point):
         if key == "outline" and data['outline'] == outline:
             edges.append(((start, end), data))
 
-    edge, data = min(edges, key=lambda edge_data: shgeo.LineString(edge_data[0]).distance(projected_point))
+    if len(edges) > 0:
+        edge, data = min(edges, key=lambda edge_data: shgeo.LineString(edge_data[0]).distance(projected_point))
+        graph.remove_edge(*edge, key="outline")
+        graph.add_edge(edge[0], node, key="outline", **data)
+        graph.add_edge(node, edge[1], key="outline", **data)
+    else:
+        # The node lies on an outline which has no intersection with any segment.
+        # We need to add a segment to connect the inserted node with the nearest available edge from
+        # an other outline.  It's the best we can do without running into networkx no path errors.
+        for start, end, key, data in graph.edges(keys=True, data=True):
+            if key == "outline":
+                edges.append(((start, end), data))
+        edge, data = min(edges, key=lambda edge_data: shgeo.LineString(edge_data[0]).distance(projected_point))
+        graph.add_edge(edge[0], node, key='segment', underpath_edges=[], geometry=shgeo.LineString([edge[0], node]))
+        graph.add_edge(node, edge[1], key='segment', underpath_edges=[], geometry=shgeo.LineString([node, edge[0]]))
```

Comment:
This works, but I wonder if we can make it simpler.  The problem is when the starting or ending point is nearest to an outline that has no segments, right?  I guess it's a hole that's so small or thin that it fits between the rows of stitches?  How about if we just pick the nearest outline that actually has segments?  That way we don't have to add a segment.

Or... maybe we can filter out those bad hole from the shape earlier on?  I mean, we can check each outline to see if it intersects with any segments, and if not, remove it from the shape.  Then we don't have to change `insert_node()` at all.

Reply:
It actually isn't as simple as that. You are right, that these are outlines that are small (or row spacing is big :D). But just taking the nearest edge will lead to unconnected start- or end node which we don't want. Plus I rather want to make sure we get the correct start position than producing unwanted jumps. Nearest edges can be far from where we actually want to be in case of other fill methods than auto fill (especially in circular fill). Does this make sense?

... Speaking of jumps, I updated this, so it segmentizes the new line segment.